src/main/assets/js/disabled-upon-submit.ts

```diff
@@ -0,0 +1,12 @@
+import { getById } from './selectors';
+
+const disableUponSubmit = (): void => {
+  const submitButton = getById('main-form-submit') as HTMLFormElement | null;
+  const form = getById('main-form') as HTMLFormElement | null;
+  const currentUrl = document.location.pathname;
+  if (form && submitButton && currentUrl === '/your-details') {
+    form.addEventListener('submit', () => submitButton.setAttribute('disabled', 'true'));
```

Comment:
can we make this a generic thing rather than just for this particular page
we're seeing a similar issue with the applyForConditionalOrder button when it's clicked twice

Reply:
Done